k2/csrc/determinize.h

```diff
@@ -781,6 +768,7 @@ class DetStateMap {
       b = symbol + 102983 * b;
       elem = elem->prev_elements[0].prev_state;
     }
+    a += elem->state_id;
```

Comment:
Two things: (1) after suggesting this I suggested a different version involving multiplying by a prime.. please use that one.  And (2) in my original suggested code there was a comment; please include that!

Reply:
Oh, sorry for the comments (have dinner just now..)

RE prime: seems there's notification sync issue on my chrome, will add, thanks